components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

```diff
@@ -541,19 +567,40 @@ public Response apisApiIdDocumentsDocumentIdPut(String apiId, String documentId,
             String contentType, String ifMatch, String ifUnmodifiedSince) {
         try {
             APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
-            Documentation documentation = DocumentationMappingUtil.fromDTOtoDocumentation(body);
             String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
 
+            Documentation oldDocument = apiProvider.getDocumentation(documentId, tenantDomain);
+            if (oldDocument == null) {
+                throw RestApiUtil.buildNotFoundException(RestApiConstants.RESOURCE_DOCUMENTATION, documentId);
+            }
+
+            if (body.getType() == DocumentDTO.TypeEnum.OTHER && StringUtils.isBlank(body.getOtherTypeName())) {
+                //check otherTypeName for not null if doc type is OTHER
+                throw RestApiUtil.buildBadRequestException("otherTypeName cannot be empty if type is OTHER.");
+            }
+
+            String sourceUrl = body.getSourceUrl();
+            if (body.getSourceType() == DocumentDTO.SourceTypeEnum.URL &&
+                    (StringUtils.isBlank(sourceUrl) || !RestApiUtil.isURL(sourceUrl))) {
+                throw RestApiUtil.buildBadRequestException("Invalid document sourceUrl Format");
+            }
+
+            //overriding some properties
+            body.setName(oldDocument.getName());
+
+            Documentation newDocumentation = DocumentationMappingUtil.fromDTOtoDocumentation(body);
             //this will fail if user does not have access to the API or the API does not exist
             APIIdentifier apiIdentifier = APIMappingUtil.getAPIIdentifierFromApiIdOrUUID(apiId, tenantDomain);
 
-            apiProvider.updateDocumentation(apiIdentifier, documentation);
+            apiProvider.updateDocumentation(apiIdentifier, newDocumentation);
             //retrieve the updated documentation
-            documentation = apiProvider.getDocumentation(documentId, tenantDomain);
-            return Response.ok().entity(DocumentationMappingUtil.fromDocumentationToDTO(documentation)).build();
+            newDocumentation = apiProvider.getDocumentation(documentId, tenantDomain);
+            return Response.ok().entity(DocumentationMappingUtil.fromDocumentationToDTO(newDocumentation)).build();
         } catch (APIManagementException e) {
-            throw new InternalServerErrorException(e);
+            String errorMessage = "Error while updating the document " + documentId + " for API : " + apiId;
+            handleException(errorMessage, e);
```

Comment:
Catching and throwing the same exception should not be done unless for a valid scenario.